modules/base/rendering/renderablenodearrow.cpp

```diff
@@ -0,0 +1,517 @@
+        }
+        const double boundingSphere = startNode->boundingSphere();
+
+        if (!useRelative) {
+            // Recompute distance (previous value was relative)
+            prop = static_cast<float>(prop * boundingSphere);
+            prop.setExponent(11.f);
+            prop.setMaxValue(1e20f);
+        }
+        else {
+            // Recompute distance (previous value was in meters)
+            if (boundingSphere < std::numeric_limits<double>::epsilon()) {
+                LERROR(fmt::format(
+                    "Start node '{}' has invalid bounding sphere", nodeName
+                ));
+                return;
+            }
+            prop = static_cast<float>(prop / boundingSphere);
+            prop.setExponent(3.f);
+            prop.setMaxValue(1000.f);
+        }
+        // @TODO (emmbr, 2022-08-22): make GUI update when min/max value is updated
+    };
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableNodeArrow::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_renderablenodearrow");
+}
+
+RenderableNodeArrow::Shading::Shading()
+    : properties::PropertyOwner({ "Shading" })
+    , enabled(ShadingEnabledInfo, true)
+    , ambientIntensity(AmbientIntensityInfo, 0.2f, 0.f, 1.f)
+    , diffuseIntensity(DiffuseIntensityInfo, 0.7f, 0.f, 1.f)
+    , specularIntensity(SpecularIntensityInfo, 0.f, 0.f, 1.f)
+{
+    addProperty(enabled);
+    addProperty(ambientIntensity);
+    addProperty(diffuseIntensity);
+    addProperty(specularIntensity);
+}
+
+RenderableNodeArrow::RenderableNodeArrow(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _start(StartNodeInfo)
+    , _end(EndNodeInfo)
+    , _color(ColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _segments(SegmentsInfo, 10, 3, 100)
+    , _invertArrowDirection(InvertInfo, false)
+    , _arrowHeadSize(ArrowHeadSizeInfo, 0.1f, 0.f, 1.f)
+    , _arrowHeadWidthFactor(ArrowHeadWidthInfo, 2.f, 1.f, 100.f)
+    , _offsetDistance(OffsetDistanceInfo, 0.f, 0.f, 1e20f)
+    , _useRelativeOffset(RelativeOffsetInfo, false)
+    , _length(LengthInfo, 100.f, 0.f, 1e20f)
+    , _useRelativeLength(RelativeLengthInfo, false)
+    , _width(WidthInfo, 10.f, 0.f, 1e11f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    addPropertySubOwner(_shading);
+    _shading.enabled = p.performShading.value_or(_shading.enabled);
+    _shading.ambientIntensity = p.ambientIntensity.value_or(_shading.ambientIntensity);
+    _shading.diffuseIntensity = p.diffuseIntensity.value_or(_shading.diffuseIntensity);
+    _shading.specularIntensity = p.specularIntensity.value_or(_shading.specularIntensity);
+
+    _start = p.startNode;
+    addProperty(_start);
+
+    _end = p.endNode;
+    addProperty(_end);
+
+    _color = p.color.value_or(_color);
+    _color.setViewOption(properties::Property::ViewOptions::Color);
+    addProperty(_color);
+
+    addProperty(Fadeable::_opacity);
```

Comment:
Could make sense to add opacity as the first property?  I think we show the list sorted by whichever gets added first